README.md

```diff
@@ -70,21 +77,60 @@ window.addEventListener('portalactivate', e => {
 });
 ```
 
-## Use cases
+If `https://example.com/` does not need the predecessor-adoption capability, then it can instead use the generic APIs for reacting to prerendering activation or other relevant changes, outlined [in that explainer](https://github.com/jeremyroman/alternate-loading-modes/blob/gh-pages/browsing-context.md#javascript-api).
 
-_See the ["Key Scenarios"](./key-scenarios.md) document for more detail on each of these, including visualizations._
+### Navigation transitions
+
+As a more realistic example, consider a page which wants to prerender a link and provide a seamless navigation transition when it's clicked. It can do this by hiding the portal until the link is clicked,
+and then animating the portal to full screen before activation.
+
+This might be written like so:
 
-- **Pre-rendering**: by loading a page in a hidden portal, it is possible to pre-render the destination page. Then, activating the portal will instantly display the pre-rendered document in the same browser tab.
+```html
+<a href="https://example.com/" class="seamless">Click me!</a>
+
+<script>
+for (const link of document.querySelectorAll('a.seamless')) {
+  const portal = document.createElement('portal');
+  portal.src = link.href;
+  portal.hidden = true;
+  portal.style = 'position: fixed; top: 0; left: 0; width: 100px; height: 100px;';
+  document.body.append(portal);
 
-  This requires some care and cooperation from both sides to deal with the restrictions on portaled content, and the transition between portaled and activated states.
+  link.onclick = async e => {
+    if (portal.state === 'closed') {
+      // The content couldn't be portaled, likely because it didn't opt-in.
+      // Let the normal link click go through.
+      return;
+    }
+
+    e.preventDefault();
+
+    // Show the portal, and animate it to the whole viewport over 300 milliseconds.
+    portal.hidden = false;
+    if (!matchMedia('(prefers-reduced-motion: reduce)').matches) {
+      await portal.animate([{ width: '100vw', height: '100vh' }], { duration: 300 }).finished;
+    }
+
+    // Once the preview is now displayed as the whole viewport, activate.
+    // This performs the instant navigation/URL bar update/etc.
+    portal.activate();
```

Comment:
Optional nit: Might be good to illustrate error handling, at least at a high level. Something like:

```
try {
  await portal.activate()
} catch (e) {
  // Conflicting navigation, we should hide the portal.
}
```

Reply:
Good point. Maybe I can eliminate the closed check even.

Reply:
The check could still be useful if the page wants to avoid a useless animation of a portal that has no chance of successfully activating. For pages that don't noticeably change before activation (e.g. offscreen prerendering), then yeah, EAFP.